pandas/tests/frame/test_indexing.py

```diff
@@ -185,6 +186,49 @@ def test_setitem_list_of_tuples(self):
         expected = Series(tuples, index=self.frame.index, name='tuples')
         assert_series_equal(result, expected)
 
+    def test_setitem_list_missing_columns(self):
```

Comment:
can you split this test so that the two cases highlighted in the issue are separate tests. (or parametrize if more appropriate) and add the issue number as a comment.

Reply:
Updated. Thanks for the feedback.